internal/api/auth/token_verifier.go

```diff
@@ -38,26 +38,92 @@ func NewOIDCTokenVerifier(ctx context.Context, cfg *OIDCConfig) (*OIDCTokenVerif
 	// Create an ID token parser, but only trust ID tokens issued to this client id.
 	return &OIDCTokenVerifier{
 		verifier: provider.Verifier(&oidc.Config{ClientID: cfg.ClientID}),
+		cfg:      cfg,
 	}, nil
 }
 
-func (t *OIDCTokenVerifier) Verify(ctx context.Context, token string) (bool, error) {
+func (t *OIDCTokenVerifier) Verify(ctx context.Context, token string) (bool, *Claims, error) {
 	// Verify token.
 	idToken, err := t.verifier.Verify(ctx, token)
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
 
 	// Extract custom claims.
 	var claims Claims
 	if err := idToken.Claims(&claims); err != nil {
-		return false, err
+		return false, nil, err
 	}
 
 	// Check that claims are verified.
 	if !claims.EmailVerified {
-		return false, nil
+		return false, nil, nil
+	}
+
+	// Parse attributes.
+	claims.Attributes, err = t.parseAttributes(idToken)
+	if err != nil {
+		return false, nil, err
+	}
+
+	return true, &claims, nil
+}
+
+// parseAttributes extracts the attributes used for access control from the token
+// based on configuration. It finds the claim in the token based on the ClaimPath
+// and ClaimPathSeparator values and filters the attrs. based on ClaimValuePrefix.
+func (t *OIDCTokenVerifier) parseAttributes(token *oidc.IDToken) ([]string, error) {
+	if !t.cfg.ABAC.Enabled {
+		return nil, nil
+	}
+
+	var data map[string]interface{}
+	if err := token.Claims(&data); err != nil {
+		return nil, err
+	}
+
+	var keys []string
+	if t.cfg.ABAC.ClaimPathSeparator != "" {
+		keys = strings.Split(t.cfg.ABAC.ClaimPath, t.cfg.ABAC.ClaimPathSeparator)
+	} else {
+		keys = []string{t.cfg.ABAC.ClaimPath}
+	}
+
+	for i, key := range keys {
+		value, exists := data[key]
```

Comment:
```suggestion
		value, ok := data[key]
```

`exists` is clear but often we just write `ok` (the "comma ok" idiom briefly mentioned in [effective go](https://go.dev/doc/effective_go#maps))

Reply:
Not really a "comma ok" case, but done ;)